pmlc/dialect/pxa/transforms/vectorize.cc

```diff
@@ -125,6 +127,119 @@ class VectorizeCandidate {
     return success();
   }
 
+  template <typename OpTy>
+  void vectorizeMathUnaryOp(OpTy op) {
```

Comment:
Please check if we can use a single function for both Unary and BinaryOp.

Maybe we should change the name of this function to something like `CreateLoopForMathOp`

Reply:
> Please check if we can use a single function for both Unary and BinaryOp.
> 
> Maybe we should change the name of this function to something like `CreateLoopForMathOp`
It seems we cannot combine Unary and Binary into a single function.
Because we define `builder.create<OpTy>(loc, element)` and `builder.create<OpTy>(loc, element1, element2)` in two if condition branches, this will cause compiling time error. For example, when `OpTy` is `math::SinOp`, it only take one operand as argument for `create` function, the another one would raise error.